### PR TITLE
Feature/highlight rightclicked node

### PR DESCRIPTION
**Pull request by slakh96**
Completes https://github.com/orgs/VerWiki/projects/1#card-55832022

### CODE REVIEW

File: frontend/src/components/Tree/Tree.jsx [hunk 1]
```diff
@@ -99,33 +108,62 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
         translate(${d.y},0)
       `
     )
-    .attr("fill", (d) => (d.data.numChildren === 0 ? "#b30000" : "#555"))
-    .attr("r", 6);
+    .attr("fill", (d) => {
+      // Separate color if a node is being viewed by the user
+      if (d.data.name === curViewingNodeID) {
+        return "#377bfa";
+      } else if (d.data.numChildren === 0) {
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Why is this variable name simply "d"? I know this comes from a previous MR but the name doesn't tell me much about what the variable is. Is there a reason or can we find a more informative name?

File: frontend/src/components/Tree/Tree.jsx [hunk 2]
```diff
@@ -99,33 +108,62 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
...
+        return "#b30000";
+      } else {
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Let's remove the else clause and just return #555

File: frontend/src/components/Tree/Tree.jsx [hunk 3]
```diff
@@ -99,33 +108,62 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
...
+        return "#555";
+      }
+    })
+    .attr("r", (d) => {
+      // Makes node bigger to help differentiate which node 
+      // is being looked at
+      if (d.data.name === curViewingNodeID) {
+        return 10;
+      } else {
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Same thing with the else clause here

File: frontend/src/components/Tree/Tree.jsx [hunk 4]
```diff
@@ -99,33 +108,62 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
...
-    .attr("x", (d) => (d.x < Math.PI === !d.children ? 6 : -6))
-    .attr("text-anchor", (d) =>
-      d.x < Math.PI === !d.children ? "start" : "end"
-    )
-    .text((node) => node.data.name + " ");
+    .attr("x", (d) => {
+      let distance = 8;
+      if (d.x < Math.PI === !d.children) {
+        return distance;
+      } else {
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Same thing here

File: frontend/src/components/Tree/Tree.jsx [hunk 5]
```diff
@@ -99,33 +108,62 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick, onRightClick) {
...
+        return -1 * distance;
+      }
+    })
+    .attr("text-anchor", (d) => {
+      if (d.x < Math.PI === !d.children) {
+        return "start";
+      } else {
```
Comment (rafiiary) [inline frontend/src/components/Tree/Tree.jsx]: Here too please :)

**Review by rafiiary — CHANGES_REQUESTED** [review]

File: frontend/src/components/Tree/Tree.jsx (hunk 1 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: name changed

File: frontend/src/components/Tree/Tree.jsx (hunk 2 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: removed

File: frontend/src/components/Tree/Tree.jsx (hunk 3 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: removed

File: frontend/src/components/Tree/Tree.jsx (hunk 4 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: removed

File: frontend/src/components/Tree/Tree.jsx (hunk 5 above)
Comment (slakh96) [reply frontend/src/components/Tree/Tree.jsx]: removed

**Review by rafiiary — APPROVED** [review]

**Pull request merged**